### PR TITLE
Java: Add `Zremrangebylex` command. (Sorted Set Group)

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 1]
```diff
@@ -1471,6 +1475,29 @@ public T zrankWithScore(@NonNull String key, @NonNull String member) {
...
+     * @param minLex The minimum lex to remove from. Can be an implementation of {@link InfLexBound}
+     *     representing positive/negative infinity, or {@link LexBoundary} representing a specific lex
+     *     and inclusivity.
+     * @param maxLex The maximum lex to remove to. Can be an implementation of {@link InfLexBound}
+     *     representing positive/negative infinity, or {@link LexBoundary} representing a specific lex
+     *     and inclusivity.
+     * @return CommandResponse - The number of members removed.<br>
+     *     If <code>key</code> does not exist, it is treated as an empty sorted set, and the command
+     *     returns <code>0</code>.<br>
+     *     If <code>minScore</code> is greater than <code>maxScore</code>, <code>0</code> is returned.
```
Comment (jonathanl-bq) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1493]: There's no minScore or maxScore here.

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 2]
```diff
@@ -456,4 +459,31 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
...
+     * @param minLex The minimum lex to remove from. Can be an implementation of {@link InfLexBound}
+     *     representing positive/negative infinity, or {@link LexBoundary} representing a specific lex
+     *     and inclusivity.
+     * @param maxLex The maximum lex to remove to. Can be an implementation of {@link InfLexBound}
+     *     representing positive/negative infinity, or {@link LexBoundary} representing a specific lex
+     *     and inclusivity.
+     * @return The number of members removed.<br>
+     *     If <code>key</code> does not exist, it is treated as an empty sorted set, and the command
+     *     returns <code>0</code>.<br>
+     *     If <code>minScore</code> is greater than <code>maxScore</code>, <code>0</code> is returned.
```
Comment (jonathanl-bq) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:478]: There's no minScore or maxScore here.

**Review by jonathanl-bq — APPROVED** [review]
A couple of small comments that need to be addressed, but otherwise LGTM.

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 3]
```diff
@@ -456,4 +459,31 @@ CompletableFuture<Map<String, Double>> zrangeWithScores(
...
+     * @example
+     *     <pre>{@code
+     * Long payload1 = client.zremrangebylex("mySortedSet", new LexBoundary("a", false), new LexBoundary("e")).get();
+     * assert payload1 == 4L; // Indicates that 4 members, with lexicographical values ranging from "a" (exclusive) to "e" (inclusive), have been removed from "mySortedSet".
+     *
+     * Long payload2 = client.zremrangebylex("nonExistingSortedSet", InfLexBound.NEGATIVE_INFINITY , InfLexBound.POSITIVE_INFINITY).get();
+     * assert payload2 == 0;
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:485]: This example is more interesting
```suggestion
     * Long payload2 = client.zremrangebylex("mySortedSet", InfLexBound.NEGATIVE_INFINITY, new LexBoundary("e")).get();
     * assert payload2 == 0; // Indicates that no elements were removed
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**